src/libs/renderer/src/sdevice.cpp

```diff
@@ -1669,6 +1676,30 @@ bool DX9RENDER::TextureLoad(long t)
     return true;
 }
 
+bool DX9RENDER::TextureLoadUsingD3DX(const char* path, long t)
+{
+    // TODO: reimplement the whole thing in a tidy way
+    IDirect3DTexture9 *pTex;
+    if(CHECKD3DERR(D3DXCreateTextureFromFileA(d3d9, path, &pTex)))
+    {
+        delete Textures[t].name;
```

Comment:
shouldn't it be delete[]?

Reply:
It should, but it's a subject for a different PR. I mentioned that in the code. This is copied from the original code and the original code is a mess.
So, let's separate a feature from existing issues. I will take care of this when I have time...